app/src/main/java/com/chouten/app/data/WebviewHandler.kt

```diff
@@ -2,358 +2,291 @@ package com.chouten.app.data
-     * This should be called before any other webview methods.
-     */
+    /** Initialize the webview handler. This should be called before any other webview methods. */
     @SuppressLint("SetJavaScriptEnabled", "JavascriptInterface")
     fun initialize(context: Context) {
         if (this::webview.isInitialized) return
+
+        WebView.setWebContentsDebuggingEnabled(true)
         webview = WebView(context)
         webview.settings.javaScriptEnabled = true
         webview.settings.domStorageEnabled = true
-        webview.addJavascriptInterface(this, "Android")
+        webview.addJavascriptInterface(this, "Native")
     }
 
-    /**
-     * destroy the webview. And reinstantiate it.
-     */
-    fun reset(context: Context) {
-        webview.destroy()
-        initialize(context)
+    fun setCallback(callback: (String) -> (Unit)){
+        this.callback = callback;
     }
 
-    fun destroy() {
-        webview.clearCache(true)
-        webview.destroy()
+    fun dontCloseOnError(){
+        this.closeOnError = false;
     }
-
-    /**
-     * Load the webview with data from the given url
-     */
-    suspend fun load(codeblock: ModuleModel.ModuleCode.ModuleCodeblock): Boolean {
-        var request = codeblock.request
-
-        // Use the client to get the data from the url
-        // and then inject it into the webview.
-        var headers = request?.headers?.mapNotNull {
-            if (it.key.isBlank() || it.value.isBlank()) return@mapNotNull null
-            it.key to it.value
-        }?.toMap()
-
-        val url = let {
-            val url = if (request?.url?.isEmpty() == true) nextUrl else request?.url ?: nextUrl
-            if (request == null || request?.url?.isEmpty() == true) request =
-                ModuleModel.ModuleCode.ModuleCodeblock.ModuleRequest(url, "GET", listOf(), null)
-            url
-        }
-        println("Requesting $url & Uses api: ${codeblock.usesApi}")
-        val data: ByteArray = try {
-            var counter = 0
-            var responseCode = -1
-            lateinit var response: ByteArray
-            while (counter < 2 || responseCode != 200) {
-                if (headers == null) headers = mapOf("User-Agent" to "Chouten")
-                val _response = when (request?.method) {
-                    "GET" -> client.get(url, headers)
-                    "POST" -> client.post(url, headers, request?.body)
-                    "PUT" -> client.put(url, headers, request?.body)
-                    "DELETE" -> client.delete(url, headers)
-                    else -> client.get(url, headers)
+    
+    @JavascriptInterface
+    fun sendHTTPRequest(data: String) {
+        val self = this
+
+        runBlocking {
+            try{ 
+                self.postMessage(data) 
+            }catch(error: Exception){
+                val message = error.message;
+
+                if(message != null){
+                    val errorJSON = mapOf<String, String>(
```

Comment:
Why not use your `ErrorJSON` data class?

Reply:
CAUSE I DON'T WANNA

Reply:
I actually don't know how to do this, I'll add a TODO, so this can be resolved later.